src/main/java/net/neoforged/neoforge/client/extensions/common/IClientItemExtensions.java

```diff
@@ -143,18 +145,23 @@ default Model getGenericArmorModel(LivingEntity livingEntity, ItemStack itemStac
      */
     default void setupModelAnimations(LivingEntity livingEntity, ItemStack itemStack, EquipmentSlot equipmentSlot, Model model, float limbSwing, float limbSwingAmount, float partialTick, float ageInTicks, float netHeadYaw, float headPitch) {}
 
+    /**
+     * @deprecated Switch to {@link IClientItemExtensions#renderHelmetOverlay(ItemStack, Player, GuiGraphics, DeltaTracker)}
+     */
+    @Deprecated(forRemoval = true)
+    default void renderHelmetOverlay(ItemStack stack, Player player, int width, int height, float partialTick) {}
+
     /**
      * Called when the client starts rendering the HUD, and is wearing this item in the helmet slot.
      * <p>
      * This is where pumpkins would render their overlay.
      *
-     * @param stack       The item stack
-     * @param player      The player entity
-     * @param width       The viewport width
-     * @param height      Viewport height
-     * @param partialTick Partial tick time, useful for interpolation
+     * @param stack        The item stack
+     * @param player       The player entity
+     * @param guiGraphics  The gui graphics
+     * @param deltaTracker The delta tracker
      */
-    default void renderHelmetOverlay(ItemStack stack, Player player, int width, int height, float partialTick) {}
+    default void renderHelmetOverlay(ItemStack stack, Player player, GuiGraphics guiGraphics, DeltaTracker deltaTracker) {}
```

Comment:
Call the deprecated method by default.
```suggestion
    default void renderHelmetOverlay(ItemStack stack, Player player, GuiGraphics guiGraphics, DeltaTracker deltaTracker) {
    renderHelmetOverlay(stack, player, guiGraphics.guiWidth(), guiGraphics.guiHeight(), deltaTracker.getGameTimeDeltaPartialTick(true));
    }
```